drivers/platform/stm32/stm32_dma.c

```diff
@@ -118,6 +119,50 @@ int stm32_dma_config_xfer(struct no_os_dma_ch *channel,
 	default:
 		return -EINVAL;
 	}
+#else
```

Comment:
I think the best play here is to wrap H5 specific code into an #if defined (STM32F5) directive, and put the existing implementation into the #else branch.

Reply:
Even better, it looks like the fudamental difference is between the DMA Cores so maybe you can find a macro that you can use to distinguish instead of distinguishing by chip family because we are not ST and we don't want to care which family has which type of DMA Core.

Reply:
Wrapping the H5 specific code into #if directive can be done; but even though it is true that F2, F4 and F7 have a different DMA type than H5, there does not seem to be any specific macro in the STM32 drivers that help us distinguish between them.